### DISCUSSION TITLE
Is it possible to get the actual endpoint callable when using a custom APIRoute class on a router.

### DISCUSSION THREAD

**Discussion by jimcarreer** [fastapi/fastapi#7797] [Questions] [answered]
### First check

* [ x] I used the GitHub search to find a similar issue and didn't find it.
* [ x] I searched the FastAPI documentation, with the integrated search.
* [ x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Is it possible to get the actual endpoint callable when using a custom APIRoute class on a router. 

### Additional context

Specifically I would like to define something like the following:

```python
import time
from typing import Callable

from fastapi import APIRouter, FastAPI
from fastapi.routing import APIRoute
from starlette.requests import Request
from starlette.responses import Response

class CustomDecorator:
    # ... standard class decorator implementation ...


class AnnotatedRoute(APIRoute):
    def get_route_handler(self) -> Callable:
        original_route_handler = super().get_route_handler()

        async def custom_route_handler(request: Request) -> Response:
            # ... get / examine the final handler signature here for CustomDecorator and do something with it ...
            response: Response = await original_route_handler(request)

        return custom_route_handler


app = FastAPI()
router = APIRouter(route_class=AnnotatedRoute)

@router.get("/timed")
@CustomDecorator(...)
async def timed():
    return {"message": "Data"}
```

I would like to use this for authorization / RBAC decoration of routes.  I've looked at the Starlet Middleware for Authentication and its not sufficient for my goals.  This was pretty trivial to do in flask so Im looking to be able to do something similar.

**Comment by jimcarreer** [comment, marked as answer]
Oh I'm an idiot, sorry I realized not 10 minutes after writing this that my custom APIRoute implementation will have the property "endpoint" which is the original callable associated with the route. Sorry for the needless chatter, that's what I get for coding drunk.

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
